plugin/pkg/shoot/validator/admission.go

```diff
@@ -215,6 +215,10 @@ func (v *ValidateShoot) Admit(ctx context.Context, a admission.Attributes, o adm
 		if reflect.DeepEqual(shoot.Spec, oldShoot.Spec) && reflect.DeepEqual(shoot.ObjectMeta, oldShoot.ObjectMeta) {
 			return nil
 		}
+
+		if !reflect.DeepEqual(shoot.Spec.SeedName, oldShoot.Spec.SeedName) {
```

Comment:
Why do we need this? 👀

Reply:
Just to print a proper error msg if somebody tries to update the seedName through shoot yaml

Reply:
Is Kubernetes doing the same?

Reply:
I don't know about that, but we can use this. If somebody tries to set the seedName to nil or another seed, the strategy will reset it, but there won't be any message. So we felt it'll be better to put this in the shoot validator, and in the binding validator, reject the change only when `SeedChange` feature gate is disabled.